### PR TITLE
Adding support for throw without catch or finally

### PR DESCRIPTION
**Pull request by TonyRice**
https://github.com/storyscript/runtime/issues/512

### CODE REVIEW

File: tests/integration/processing/Lexicon.py [hunk 1]
```diff
@@ -972,6 +972,15 @@
...
                      key='arr', expected=[1, 42, 42, 3, 4, 4, 5, 5])),
         ]),
+    Suite(
+        preparation_lines='failed = true\n'
+                          'try\n'
+                          '  throw "error"\n',
+        cases=[
+            Case(append='failed = false',
+                 assertion=ContextAssertion(key='failed', expected=False))
+        ]
```
Comment (wilzbach) [inline tests/integration/processing/Lexicon.py]: Add this too:

```
    Suite(
        preparation_lines='failed = true\n'
                          'try\n'
                          '  a = 1\n',
        cases=[
            Case(assertion=ContextAssertion(key='failed', expected=True))
        ]
    ),
    Suite(
        preparation_lines='failed = true\n'
                          'try\n'
                          '  throw "error"' 
                          '  failed = false\n'
        cases=[
            Case(assertion=ContextAssertion(key='failed', expected=True))
        ]
    ),

```

File: storyruntime/processing/Lexicon.py [hunk 2]
```diff
@@ -403,7 +403,10 @@ def unless_condition(logger, story, line):
             if LineSentinels.is_sentinel(exec_res):
                 result_sentinel = exec_res
         except StoryscriptError:
-            if next_line['method'] == 'finally':
+            if next_line['method'] != 'finally' and \
+                    next_line['method'] != 'catch':
+                return Lexicon.line_number_or_none(next_line)
```
Comment (adnrs96) [inline storyruntime/processing/Lexicon.py]: I don't think we wanna do this over here. Thing is `try` block had an error and I believe if we can't handle it over here we need to raise it upstairs for a higher try catch to handle.
Thoughts?

File: tests/integration/processing/Lexicon.py [hunk 3]
```diff
@@ -1753,6 +1753,33 @@
...
+            Case(assertion=ContextAssertion(key='a', expected=None))
+        ]
+    ),
+    Suite(
+        preparation_lines='failed = true\n'
+                          'try\n'
+                          '  throw "error"\n'
+                          '  failed = false\n',
+        cases=[
+            Case(assertion=ContextAssertion(key='failed', expected=True))
```
Comment (adnrs96) [inline tests/integration/processing/Lexicon.py]: We should test that the error thrown was with message `error` over here

File: storyruntime/processing/Lexicon.py (hunk 2 above)
Comment (adnrs96) [reply storyruntime/processing/Lexicon.py]: I would definitely like to hear about what you think about this

File: tests/integration/processing/Lexicon.py (hunk 3 above)
Comment (TonyRice) [reply tests/integration/processing/Lexicon.py]: On it!

**Comment by adnrs96** [comment]
Had a chat with Tony and things make sense as it is.

File: tests/integration/processing/Lexicon.py [hunk 4]
```diff
@@ -857,8 +857,8 @@
             ),
             Case(
                 append=f'a = {math.pi / 4}\n'
-                       'b = a.tan()\n'
-                       'c = b.approxEqual(value: 1)',
+                'b = a.tan()\n'
+                'c = b.approxEqual(value: 1)',
```
Comment (wilzbach) [inline tests/integration/processing/Lexicon.py]: UNRELATED CHANGE!

Comment (anukul) [reply tests/integration/processing/Lexicon.py]: 😭😭😭

Comment (TonyRice) [reply tests/integration/processing/Lexicon.py]: This change keeps autoformatting. I'll undo that.
